File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curr = new_head = ListNode(-1)
        val = 0
        head = head.next
        while(head):
            val += head.val
            if(not head.val):
                new_node = ListNode(val)
                curr.next = new_node
                val = 0
                curr = new_node
                
            head = head.next
        
        return new_head.next
            

File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeNodes(head *ListNode) *ListNode {
    var curr, ret, prev = head, head, &ListNode{}
    head = head.Next
    curr.Next = nil
    for head != nil {
        if head.Val == 0 {
            prev.Next = curr
            curr.Next = nil
            prev = curr
            curr = head
        }
        curr.Val += head.Val
        head = head.Next
    }
    
    return ret
}